### ISSUE TITLE
VFEmai.netl isn't a supported email domain

### ISSUE THREAD

**Issue by Zero3K** [bluesky-social/social-app#5981]
### Steps to Reproduce

1. Go to Settings.
2. Try to change the email address to one that uses the vfemail,net domain.

### What platform(s) does this occur on?

iOS, Android, Web (Desktop), Web (Mobile)

### What version of the app are you using?

1.92.1

**Comment by surfdude29** [comment]
What error message are you seeing? I just tried to reproduce this (by entering a nonsense email address ending in @vfemail.net) and I got to the next step fine.

**Comment by Zero3K** [comment]
This email address is not supported, please use a different email.

**Comment by surfdude29** [comment]
> This email address is not supported, please use a different email.

Out of interest, do you see that error message as soon as you've typed in your email address and tapped **Request Change**, or only after you've typed in the confirmation code?

In any case, from what I can tell I think you're right that `vfemail.net` isn't a supported email domain. (But to be clear, you'll have to email support or wait for one of the team to respond here for an official answer.)

The error message you're seeing comes from here:

https://github.com/bluesky-social/atproto/blob/main/packages/pds/src/api/com/atproto/server/updateEmail.ts#L19

and it looks like the email address is not being validated because it's considered to be 'disposable'. Bluesky seems to use [this package](https://www.npmjs.com/package/disposable-email) to screen out some email domains, and unfortunately `vfemail.net` seems to be considered disposable by that package as it's on this list (⚠️ warning, large file):

https://raw.githubusercontent.com/disposable/disposable-email-domains/master/domains.txt

So I'm afraid it looks like you'll have to use another email address.

**Comment by Zero3K** [comment]
VFEmail isn't disposable though since it costs money to have a decent amount of storage and features.

**Comment by Zero3K** [comment]
I made an issue on that list's repository. It can be found at https://github.com/disposable/disposable/issues/196.

**Comment by Zero3K** [comment]
My pull request to fix it (https://github.com/disposable/disposable/pull/197) has been merged. Now to wait for the domains.txt to be updated (which will no longer have vfemail.net in it once it is updated).

**Comment by Zero3K** [comment]
Weird, the domain is removed from that list but Bluesky is still refusing it. I guess Bluesky is using an outdated list.

**Comment by Zero3K** [comment]
What can be done to have it use the latest domains.txt?

**Comment by Zero3K** [comment]
Hmm?

**Comment by surfdude29** [comment]
> What can be done to have it use the latest domains.txt?

I think the issue might be that Bluesky seems to be using the npm package rather than the raw list of domains, and the npm package hasn't been updated for 8 years... 😬

**Comment by Zero3K** [comment]
That needs fixing then.

**Comment by surfdude29** [comment]
I just noticed that this has been merged: https://github.com/bluesky-social/atproto/pull/3092 which changes the packages used to validate email addresses. So when that has been rolled out – this is just a guess, but it could be anywhere from a few days to a week, but it could be longer, of course – you might want to try again, as it's possible `vfemail.net` will be supported 🤞

(Note that I haven't looked into the new packages being used to check, the info is in that PR if you want to check for yourself :)

**Comment by Zero3K** [comment]
I made a new PR on the repository that has the list that the PR is having Bluesky use. It is at https://github.com/disposable-email-domains/disposable-email-domains/pull/538.

**Comment by Zero3K** [comment]
How can I know if it has been rolled out?

**Comment by surfdude29** [comment]
The only way I know – there may be an easier way of course – is to look at what version the Bluesky PDSes are running here:

https://github.com/mary-ext/atproto-scraping

And then look at that git commit.

But because `vfemail.net` is in the `disposable-email-domains` package, I wouldn't get your hopes up 🤷🏻‍♂️

**Comment by Zero3K** [comment]
The vfemail.net domain should be removed from the list the package uses soon. Just need to wait for the guy who can commit the PR I made to see the screenshots I replied with in the issue I made regarding it.

**Comment by Zero3K** [comment]
Weird, the change has been rolled out and the PR I made has been merged but the domain is still being blocked.

**Comment by Zero3K** [comment]
I think it is because it is using an outdated blocklist. So, it is going to have to be updated when a new version of that package is available. I told the guy who develops it to update its blocklist and made an issue for getting it to always using the latest one.

**Comment by Zero3K** [comment]
I made a PR to update the package version used since it was updated to use the latest .conf file. It can be found at https://github.com/bluesky-social/atproto/pull/3138. Merging it should get this issue to be fixed after the scraper shows that it has been updated.

**Comment by Zero3K** [comment]
It hasn't been merged yet.
